cluster-autoscaler/cloudprovider/gce/templates.go

```diff
@@ -228,18 +231,65 @@ func (t *GceTemplateBuilder) BuildNodeFromTemplate(mig Mig, template *gce.Instan
 	return &node, nil
 }
 
-// isEphemeralStorageWithInstanceTemplateDisabled will allow bypassing Disk Size of Boot Disk from being
+func ephemeralStorageLocalSSDCount(kubeEnvValue string) int64 {
+	v, found, err := extractAutoscalerVarFromKubeEnv(kubeEnvValue, "ephemeral_storage_local_ssd_count")
+	if err != nil {
+		klog.Warningf("cannot extract ephemeral_storage_local_ssd_count from kube-env, default to 0: %v", err)
+		return 0
+	}
+
+	if !found {
+		return 0
+	}
+
+	n, err := strconv.Atoi(v)
+	if err != nil {
+		klog.Warningf("cannot parse ephemeral_storage_local_ssd_count value, default to 0: %v", err)
+		return 0
+	}
+
+	return int64(n)
+}
+
+func getLocalSSDEphemeralStorageFromInstanceTemplateProperties(instanceProperties *gce.InstanceProperties, ssdCount int64) (ephemeralStorage int64, err error) {
+	if instanceProperties.Disks == nil {
+		return 0, fmt.Errorf("instance properties disks is nil")
+	}
+
+	var count int64
+	var diskSizeGiB int64
+	for _, disk := range instanceProperties.Disks {
+		if disk != nil && disk.InitializeParams != nil {
+			if disk.Type == "SCRATCH" && disk.InitializeParams.DiskType == "local-ssd" {
+				count++
+				if diskSizeGiB == 0 {
+					diskSizeGiB = disk.InitializeParams.DiskSizeGb
+				} else if diskSizeGiB != disk.InitializeParams.DiskSizeGb {
+					return 0, fmt.Errorf("local SSDs of different sizes are not supported")
+				}
```

Comment:
In your proposal, it's okay if the template has ssdCount disks of size LOCAL_SSD_DISK_SIZE, and N disks of any sizes. I guess that's all we need. The check I had, "local SSDs of different sizes are not supported", is indeed too strict, and would allow other disk sizes for which experimental filesystem overheads have not been measured. Anyway, in practice, all disks are 375GB at the moment, so the checks are equivalent, but I'll switch to yours.